caffe2/serialize/crc_alt.h

```diff
@@ -0,0 +1,1332 @@
+#pragma once 
+
+// //////////////////////////////////////////////////////////
+// Crc32.h
+// Copyright (c) 2011-2019 Stephan Brumme. All rights reserved.
+// Slicing-by-16 contributed by Bulat Ziganshin
+// Tableless bytewise CRC contributed by Hagai Gold
+// see http://create.stephan-brumme.com/disclaimer.html
```

Comment:
From the site: 

> Unless otherwise noted, all source code published on http://create.stephan-brumme.com and its sub-pages is licensed similar to the zlib license:
> 
> This software is provided 'as-is', without any express or implied warranty. In no event will the author be held liable for any damages arising from the use of this software.
> Permission is granted to anyone to use this software for any purpose, including commercial applications, and to alter it and redistribute it freely, subject to the following restrictions:
> The origin of this software must not be misrepresented; you must not claim that you wrote the original software.
> If you use this software in a product, an acknowledgment in the product documentation would be appreciated but is not required.
> Altered source versions must be plainly marked as such, and must not be misrepresented as being the original software.
> If you like / hate / ignore my software, send me an email or, even better, a nice postcard. Thank you ! ☺

Should be fine to use in pytorch. I do not plan on moving this to 3rd party as I have edited some of the preprocessor macros to work better for our needs.

Reply:
if you haven't already can you mark the places where you edited it?